dotnet/src/SemanticKernel.Skills/Skills.Compression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.Compression;
+
+//**********************************************************************************************************************
+// EXAMPLE USAGE
+//
+// ISemanticKernel kernel = SemanticKernel.Build();
+// var zipCompressor = new ZipFileCompressor();
+// var skill = new FileCompressionSkill(zipCompressor);
+// var fileCompression = kernel.ImportSkill(skill, "FileCompression");
+// string sourceFilePath = "FileToCompress.txt";
+// string destinationFilePath = "CompressedFile.zip";
+// var variables = new ContextVariables(sourceFilePath);
+// variables.Set(FileCompressionSkill.Parameters.DestinationFilePath, destinationFilePath);
+// await kernel.RunAsync(variables, fileCompression["CompressFileAsync"]);
+//**********************************************************************************************************************
+
+/// <summary>
+/// Skill for compressing and decompressing files.
+/// </summary>
+public class FileCompressionSkill
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Directory to which to extract compressed file's data.
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// File path where to save compressed data.
+        /// </summary>
+        public const string DestinationFilePath = "destinationFilePath";
+    }
+
+    private readonly IFileCompressor _fileCompressor;
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressFileAsync(string sourceFilePath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressFileAsync)} got called");
```

Comment:
perf: let the logger interpolate the string. Similar comment on the other log calls

Reply:
Are you asking to do something like the following even though the only interpolated argument is a constant? (And is this probably resolved at compile time)
```suggestion
        this._logger.LogDebug($"{methodName} got called", nameof(CompressFileAsync));
```

Reply:
Actually, this is probably a moot point now with the other changes regarding the re-use of the logged string.

Reply:
```csharp
this._logger.LogDebug($"{0} got called", nameof(CompressFileAsync));
```

Reply:
That's what I understood.  I'll do that if you ask me to do it, but you mentioned performance...  Then why would we want the logger to do this at runtime when the interpolation would be done as compile time (if we're using C# 10 or above) since it's dealing with a constant?

Reply:
my bad I didn't notice it was a constant (pls don't do it just because someone asks :-)). I think we're on the same page, but just in case:
* if the interpolation happens at build time, good!
* if the interpolation happens at runtime, defer to the logger.

I'll check the other calls